[XinghuaJulia] iP

XinghuaJulia opened the pull request:
# Ai
Your personal Ai to provide you with 愛

### So... What can ~~my~~ your beloved Ai do?

- Dance
- Sing: have you heard her **best** [song](https://www.youtube.com/watch?v=IIHoq68i5f0)?
- Comfort you when you are down (which happens very often)

### How to get your personal Ai **NOW**???
_C'est facile!_

1. Go to your nearest GameStop
2. Buy some DVDs
3. Cry
4. Cry a lot
5. I forgot.


### Demo
Your Ai can keep track of tasks for you

- [ ] buy gifts for Ai's birthday 🎁 
- [ ] hangout with Ai

> Lies are the most exquisite love! 😉

```
System.out.println("愛してる");
sendLove.execute();
```

Surely, `ai.isLove` must be true...

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Rename, Greet, Exit

Commit:
Echo

Commit:
Add, List

Commit:
Mark as Done

Commit:
ToDos, Events, Deadlines

Commit:
Automated Text UI Testing

Commit:
Handle Errors

Commit:
Delete, Use Java Collections Classes

Commit:
Add enums for the different cases

Commit:
Merge branch 'master' of https://github.com/XinghuaJulia/ip into branch-Level-7

Commit:
Add file read and safe to retain data between each bootup of Ai

Commit:
Merge branch 'branch-Level-7'

Commit:
Add date and time, stores date and time as java.time.LocalDate

Commit:
Merge branch 'branch-Level-8'

Commit:
Add more classes to adhere to OOP principles

Commit:
Merge branch 'branch-A-MoreOOP'

Commit:
Organise classes into packages: ai, ai.command, ai.task, ai.exception

Commit:
Merge branch 'branch-A-Packages'

Commit:
Merge branch 'add-gradle-support'

Commit:
Merge branch 'master' of https://github.com/XinghuaJulia/ip into add-gradle-support

Commit:
Integrade gradle support

Commit:
Add Deadline and ToDo JUnit tests

Commit:
Merge branch 'add-gradle-support'

Commit:
Add comments to most public methods and classes

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Reformat code using IntelliJ

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Add FindCommand to find the task containing the keyword

Commit:
Merge branch 'branch-Level-9'

Commit:
Add Checkstyle.xml and Suppressions.xml configuration

Commit:
Fix style violations

Commit:
Add GUI to Ai

Commit:
Fix formatting issue for due function

Commit:
Merge branch 'branch-Level-10'

gohqingkhang commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,30 @@
+import java.io.IOException;
```
Maybe can consider placing this class into a package? Same applies for DialogBox.java, Launcher.java, etc :)

gohqingkhang commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,30 @@
+import javafx.stage.Stage;
+
+/**
+ * A GUI for Duke using FXML.
```
Should this GUI be for Duke?

XinghuaJulia replied:
XD You're right, thanks!

gohqingkhang commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,12 @@
 public class Duke {
```
Should this class name still be Duke?

XinghuaJulia replied:
I have removed it altogether :p

gohqingkhang commented on src/main/java/DialogBox.java:
```diff
@@ -0,0 +1,63 @@
+        return new DialogBox(text, img);
+    }
+
+    public static DialogBox getDukeDialog(String text, Image img) {
```
Maybe the name for this method should be getAiDialog?

gohqingkhang commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,30 @@
+            AnchorPane ap = fxmlLoader.load();
+            Scene scene = new Scene(ap);
+            stage.setScene(scene);
+            fxmlLoader.<MainWindow>getController().setAi(ai); // inject the Duke instance
```
Maybe the comments for this line could be changed?

gohqingkhang commented on src/main/java/MainWindow.java:
```diff
@@ -0,0 +1,52 @@
+    private Ai ai;
+
+    private Image userImage = new Image(this.getClass().getResourceAsStream("/images/User.png"));
+    private Image dukeImage = new Image(this.getClass().getResourceAsStream("/images/Ai.png"));
```
Maybe the name for this image could be changed?

gohqingkhang commented on src/main/java/ai/Ai.java:
```diff
@@ -0,0 +1,78 @@
+    public void run() {
+        System.out.println(ui.showGreetings());
+
+        Scanner scanner = new Scanner(System.in); //Create a Scanner object
```
Maybe a space after // in the comments could be good?

gohqingkhang commented on src/main/java/ai/Storage.java:
```diff
@@ -0,0 +1,103 @@
+     */
+    public ArrayList<Task> load() throws IOException, AiException {
+        Path path = Paths.get(filePath);
+        Path directory = path.getParent();
```
Maybe the name can be more descriptive, not to be mistaken with the actual directory file

```suggestion
        Path directoryPath = path.getParent();
```

gohqingkhang commented on src/main/java/ai/Storage.java:
```diff
@@ -0,0 +1,103 @@
+     * @return Task in the correct type.
+     * @throws AiException if the lines cannot be read.
+     */
+    public Task readFileLine(String input) throws AiException {
```
Good that you extracted the process of reading file lines into a separate method!

gohqingkhang commented on src/main/java/ai/TaskList.java:
```diff
@@ -0,0 +1,90 @@
+     *
+     * @param date Date to be matched.
+     */
+    public String isDue(LocalDate date) {
```
Maybe the name of this method can be changed? as isDue seems to imply that the return type is a boolean :)

```suggestion
    public String getDueTasks(LocalDate date) {
```

gohqingkhang commented on src/main/java/ai/task/Task.java:
```diff
@@ -0,0 +1,68 @@
+    }
+
+    /**
+     * Creates a new Task object.
+     * @param description String to be stored as description.
+     * @param isDone whether the task is done.
```
Maybe there should be an empty line between the description and the parameters? Same applies for all other JavaDoc comments :)

gohqingkhang commented on src/main/java/ai/task/Task.java:
```diff
@@ -0,0 +1,68 @@
+    /**
+     * Formats the Task for writing into the .txt file.
+     *
+     * @return String.
```
Maybe the return comment can be more descriptive on what kind of String?

gohqingkhang reviewed (commented):
Overall LGTM! Just minor nits in naming conventions and coding standards :)

CreditableX commented on src/main/java/ai/Parser.java:
```diff
@@ -0,0 +1,81 @@
+import ai.exception.AiException;
+import ai.task.Deadline;
+import ai.task.Event;
+import ai.task.ToDo;
```
Nice importing instead of using *

CreditableX commented on src/main/java/ai/Ui.java:
```diff
@@ -0,0 +1,42 @@
+    /**
+     * Prints the first line user sees when booting up Ai.
+     */
+    public String showGreetings() {
```
Maybe change to showGreeting to be in line with the standard?

XinghuaJulia replied:
Noted, thank you!!

CreditableX commented on src/test/java/task/DeadlineTest.java:
```diff
@@ -0,0 +1,66 @@
+    }
+
+    @Test
+    public void deadline_garbageInput_exceptionThrown() {
```
Maybe garbageInput could be changed to wrongDateTimeFormat for better clarity - there's one below but it might be good to have two?

XinghuaJulia replied:
Great idea! Thanks a lot! Garbage input does sound funny and ambiguous :/

CreditableX commented on src/test/java/task/DeadlineTest.java:
```diff
@@ -0,0 +1,66 @@
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class DeadlineTest {
+    @Test
```
Good naming of tests!

Commit:
Add assert in certain parts

XinghuaJulia committed:
Update src/main/java/ai/Storage.java

Co-authored-by: gohqingkhang <143182152+gohqingkhang@users.noreply.github.com>

Commit:
Change Parser and add EmptyArgumentAiException to adhere to code quality

Commit:
Change Parser and add EmptyArgumentAiException to adhere to code quality

Commit:
Replace switch statement in Storage with Java Stream

Commit:
Replace switch statement in Storage with Java Stream

Commit:
Merge branch 'branch-A-Assertions'

Commit:
Merge branch 'master' of https://github.com/XinghuaJulia/ip

Commit:
Revert "Merge branch 'master' of https://github.com/XinghuaJulia/ip"

This reverts commit 496ce93523d471b167fafa5585fa46e483dc4d53, reversing
changes made to 661f607b5b4779efa57dc11b64b17debbd77bb7f.

Commit:
Revert "Merge branch 'branch-A-Assertions'"

This reverts commit 661f607b5b4779efa57dc11b64b17debbd77bb7f, reversing
changes made to 38f6d58008237b594d961f419495d571b208efdc.

Commit:
Merge branch 'master' of https://github.com/XinghuaJulia/ip

XinghuaJulia committed:
Merge pull request #2 from XinghuaJulia/branch-A-CodeQuality

Change Parser and add EmptyArgumentAiException to adhere to code quality

Commit:
Resolve merge conflict at Storage

XinghuaJulia committed:
Merge pull request #3 from XinghuaJulia/branch-A-Streams

Replace switch statement in Storage with Java Stream

Commit:
Implement feature to prevent adding duplicate tasks

Deal with duplicate items

Tasks of the same type with same fields (ToDo, Event, Deadline) can
currently be added multiple times without being marked as duplicates.

This can cause duplicate tasks, causing users to have a difficult time
managing their tasks.

Let's introduce the ability to recognise duplicate tasks
* Add an equals() method for the Task class and its subclasses.
* Iterate through the tasks Tasklist using equals() method to detect
  duplicates before adding new tasks.

This improves user experience by preventing redundant tasks from being
added.

Commit:
Implement feature to prevent adding duplicate tasks

Commit:
Merge branch 'branch-BCD-Extension'

Commit:
Implement Continuous Integration via yml file

Commit:
Implement Continuous Integration via yml file

Commit:
Merge branch 'branch-A-CI'

Set up Continuous Integration

This project lacks a continuous Integration setup, which is useful for
automating testing and ensuring code quality.

Without Continuous Integration, developers need to run tests manually,
increasing chances of bugs not being caught.

Let's add GitHub Actions workflow to set up CI for the project. Remove
the last three segments from the original .yml file as the tests are not
relevant.

Using Continuous Integration ensure that code is automatically tested
with each commit.

Commit:
Merge branch 'master' of https://github.com/XinghuaJulia/ip

Commit:
Tweak GUI to fit the idol aesthetic

Commit:
Merge branch 'branch-A-BetterGui'

Commit:
Add personality to Ai chatbot for some of the dialogue options

Commit:
Merge branch 'branch-A-Personality'

Commit:
Change output on how empty fields are handled

Commit:
Fix deadline empty arguments formatting issue

Commit:
Merge branch 'branch-A-Personality'

Commit:
Fix deadline empty arguments showing empty line error

Commit:
Merge branch 'branch-A-Personality'

XinghuaJulia committed:
Update README.md using ChatGPT

The current Ai user guide is functional but lacks the necessary flair to align with the playful personality of Ai. Additionally, the formatting can be improved for clarity and consistency.

This commit enhances the user guide in the following ways:

* Improves the overall grammar and readability.
* Adds playful language and personality inspired by Ai Hoshino.
* Clarifies formatting for better user experience by using clear section headers and code block examples.
* Includes a reminder about duplicate task detection, making it more intuitive for the users.

This change was made to enhance user engagement, making the guide both informative and fun to follow, matching Ai's persona.

Commit:
Add scene title to display Ai in the ai.jar file window

Commit:
Merge branch 'master' of https://github.com/XinghuaJulia/ip

Commit:
Add Ui.png file to show screenshot of product

Commit:
Fix checkstyle errors in import statements

XinghuaJulia committed:
Update README.md now has more colors

Commit:
Fix import checkstyle error in DeadlineTest and ToDoTest

Commit:
Fix empty arguments bugs for Due, Mark, Unmark, Delete, Deadline, Event

Commit:
Merge branch 'master' of https://github.com/XinghuaJulia/ip

Commit:
Merge branch 'branch-handlingEdgeCases'

Commit:
Change order of import statement in DeadlineTest

Commit:
Rearrange import statements in Test classes so that static imports come first

Commit:
Fix "todo             " as acceptable input command

Commit:
Fix final bugs and prepare for release of .jar file

XinghuaJulia committed:
Update README.md to increase picture size

XinghuaJulia committed:
Update README.md to be colourful
